### PR TITLE
improve Refgenie documentation

### PR DESCRIPTION
**Pull request by mirpedrol**
Improving refgenie documentation with examples.

⚠️ I'm not sure about adding the link to the nf-core refgenie server ([here](https://github.com/nf-core/nf-co.re/pull/1240/files#diff-53c0250c0760685395a97508dae235e6a0a305ab658041b158dec4e58dbd4e36R66-R67:~:text=%3E%20NOTE%3A%20You,igenomes.databio.org/%60)). This is not finished, and it doesn't contain all required files, which is part of the reasons why we haven't implemented the tools to use refgenie as pipeline developers. 

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1240"></a>

### CODE REVIEW

File: markdown/usage/reference_genomes.md [hunk 1]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
 
 You can also use the reference genome manager [Refgenie](http://refgenie.databio.org/en/latest/overview/) with nf-core pipelines.
 
-After installing and initializing Refgenie, an nf-core plugin will generate a configuration file which will contain the links to all downloaded reference genomes.
+1. Install and initialize refgenie following the official [documentation](http://refgenie.databio.org/en/latest/install/).
+
+A file required by nf-core containing refgenie genome assets will be automatically created at `~/.nextflow/nf-core/refgenie_genomes.config`
+An `includeConfig` statement to this new config file will be added at `~/.nextflow/config`
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:62]:
```suggestion
An `includeConfig` statement to this new config file will be appended to the `~/.nextflow/config` file. This file is automatically loaded by Nextflow during every pipeline run (see the [Nextflow documentation](https://nextflow.io/docs/latest/config.html)).
```

File: markdown/usage/reference_genomes.md [hunk 2]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
...
 
 This file should never be updated manually. To use a new reference genome or asset, by using `refgenie pull` the nf-core plugin will automatically update the configuration file.
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:64]:
```suggestion
To use a new reference genome or asset, fetch it via normal refgenie usage (`refgenie pull`) - the nf-core plugin will automatically update the `refgenie_genomes.config` configuration file.
This file should never be edited manually, as it is overwritten during each refgenie command.
```

File: markdown/usage/reference_genomes.md [hunk 3]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
...
 
+> NOTE: You can add additional refgenie servers to your refgenie configuration yaml file. Nf-core provides a [server](http://igenomes.databio.org/) with all genome files used in its pipelines.
+> Use the following command: `refgenie subscribe -s http://igenomes.databio.org/`
+
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:68]: Not needed I think. The intention with the igenomes server is to integrate at pipeline-dev level, not user level. End users should not need to have refgenie installed (or know of its existence) in order to use those genomes. I think mentioning it here will only lead to confusion.

```suggestion
```

File: markdown/usage/reference_genomes.md [hunk 4]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
...
+2. Pull all the reference assets that you may need to run the pipeline.
+
+```bash
+refgenie pull t7/fasta
+refgenie pull t7/bowtie2_index
+```
+
+Asset paths are automatically added to `~/.nextflow/nf-core/refgenie_genomes.config`.
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:76]:
```suggestion
Asset paths are automatically added to `~/.nextflow/nf-core/refgenie_genomes.config`, included in `~/.nextflow/config` and available to every pipeline run.
```

File: markdown/usage/reference_genomes.md [hunk 5]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
...
+
+The file format mimics the igenomes.config file:
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:78]: This'll need updating soon I guess! 😅 That's fine though..
```suggestion
The file format mimics the `igenomes.config` file that comes with many nf-core pipelines:
```

File: markdown/usage/reference_genomes.md [hunk 6]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
...
+// To make changes here, use refgenie to update the reference genome data
+params {
+  genomes {
+    't7' {
+      bowtie2_index        = "<path to refgenie genomes>/alias/t7/bowtie2_index/default/t7"
+      fasta                = "<path to refgenie genomes>/alias/t7/fasta/default/t7.fa"
+    }
+  }
+}
+```
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:91]:
````suggestion
```

Here, the genome _key_ that you'll use to launch the pipeline is `t7`.
````

File: markdown/usage/reference_genomes.md [hunk 7]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
...
+
+> NOTE: You can also use [custom assets](http://refgenie.databio.org/en/latest/custom_assets/).
+
+3. Run your pipeline specifying the required genome.
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:95]:
```suggestion
3. Run your pipeline, specifying the required genome.
```

File: markdown/usage/reference_genomes.md [hunk 8]
```diff
@@ -56,8 +56,46 @@ Read the [Nextflow configuration documentation](configuration.md) for more infor
...
+
+```bash
+nextflow run nf-core/<PIPELINENAME> --input samplesheet.csv --outdir <OUTDIR> --genome t7 -profile <docker/singularity/podman/shifter/charliecloud/conda/institute>
```
Comment (ewels) [inline markdown/usage/reference_genomes.md:98]:
```suggestion
nextflow run nf-core/<PIPELINENAME> --genome t7  # ..rest of normal pipeline flags..
```

**Review by ewels — APPROVED** [review]
Great! Some minor suggestions but looks good 👍🏻

**Comment by mirpedrol** [comment]
Thanks for the review!

**Pull request merged**